docs/ref/databases.txt

```diff
@@ -171,6 +171,19 @@ If you need to add a PostgreSQL extension (like ``hstore``, ``postgis``, etc.)
 using a migration, use the
 :class:`~django.contrib.postgres.operations.CreateExtension` operation.
 
+Server side cursors
+-------------------
+
+When using :ref:`queryset_iterator`, Django opens a :ref:`server-side cursor
+<psycopg2:server-side-cursors>`. By default, when PostgreSQL plans a cursor
+query, it assumes that only the first 10% of the results will be fetched. This
+results in a fast-start plan, but can take excessive time if the whole results
```

Comment:
I took the "fast-start" term from RoadiumToad in Florian's response. From what I understand, the query planner prefers to start returning results faster rather than taking time to plan the most efficient way to get all results.

I think it would be clearer to say:
By default, when PostgreSQL plans a cursor query, it assumes that only the first 10% of the results will be fetched. The query planner will spend less time planning the query and start returning rows faster, but this could diminish the query performance if more than 10% of the results are retrieved.